src/ui/components/components.rs

```diff
@@ -159,6 +165,64 @@ pub struct Chat {
     pub chat_element: gtk4::Box,
     selected_channel_id: Option<String>,
 }
+pub struct Guild {
+    guild_name: String,
+    guild_id: String,
+    guild_element: Button,
+}
+pub struct Guilds {
+    guilds: Vec<Guild>,
+    chat_stack: Stack,
+    pub guilds_element: gtk4::Box,
+    pub discriminator: u8,
+}
+impl Guilds
+{
+    pub(crate) fn new(chat_stack: Stack) -> Self {
+        let guilds_element = gtk4::Box::new(Orientation::Vertical, 0);
+        let dm_box = gtk4::Box::new(Orientation::Vertical, 0);
+
+        let discord_logo = Image::from_file(Path::new("src/ui/assets/discord_logo.png").to_owned());
+
+        dm_box.set_valign(Align::Center);
+        dm_box.append(&discord_logo);
+
+        let dm_button = Button::new();
+        dm_button.set_child(Some(&dm_box));
+
+        dm_button.connect_clicked(glib::clone!(
```

Comment:
The following clone macro has been deprecated, and is more confusing then the current practice.